src/frontend/screens/Library/components/InstallModal/index.tsx

```diff
@@ -81,7 +81,7 @@ function getInstallLanguage(
       return foundAvailable
     }
   }
-  return availableLanguages[0]
+  return availableLanguages[0]!
```

Comment:
What is the name of this feature of adding an exclamation mark at the end? I don't remember using this one before.

Reply:
It's the [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator). It's basically a way of telling Typescript "this thing can't be undefined/null right now", in the rare case that it's not able to detect this on its own

Reply:
Yes, true. I don't really use it that much because it has some issues imo:
- Adding this operator moves the responsibility to know if the prop can be null/undefined from TS to the developer.
- With that, we might actually cause something to fail because some argument or prop come null
- I like to deal with this like this is using `if` or optional chaining (not the most beautiful way but the safest imo:
`if(prop){
return prop
}`

or 

`if(something && something.value)` or `if(something?.value){`

The thing is that on this refactor there are dozens of cases where we needed this operator so might be a sign that we should maybe review some of our types or add more checks.

Reply:
but I am not arguing against using it, it is actually pretty useful, we just need to be extra careful :P

Reply:
I definitely agree that it should be the last thing you use, but there are simply cases where you *have* to use it since an undefined/null simply *can't* happen, and adding checks everywhere for situations that can't happen is pointless IMO

As a simple example, we will always have a BrowserWindow open, but `BrowserWindow.getAllWindows()` returns an array that can technically be empty. In cases like these, where it's *absolutely certain* that the given value isn't undefined/null, it's fine to use. There are just simply cases in a big project like this where the developer *can* know better than TypeScript, and the operator is meant for those situations